packages/squiggle-lang/src/rescript/Distributions/GenericDist/GenericDist.res

```diff
@@ -154,140 +154,141 @@ let truncate = Truncate.run
-  let runConvolution = (
-    toPointSet: toPointSetFn,
-    arithmeticOperation: Operation.convolutionOperation,
-    t1: t,
-    t2: t,
-  ) =>
-    E.R.merge(toPointSet(t1), toPointSet(t2))->E.R2.fmap(((a, b)) =>
-      PointSetDist.combineAlgebraically(arithmeticOperation, a, b)
-    )
-
-  let runMonteCarlo = (
-    toSampleSet: toSampleSetFn,
-    arithmeticOperation: Operation.algebraicOperation,
-    t1: t,
-    t2: t,
-  ): result<t, error> => {
-    let fn = Operation.Algebraic.toFn(arithmeticOperation)
-    E.R.merge(toSampleSet(t1), toSampleSet(t2))
-    ->E.R.bind(((t1, t2)) => {
-      SampleSetDist.map2(~fn, ~t1, ~t2)->E.R2.errMap(x => DistributionTypes.OperationError(x))
-    })
-    ->E.R2.fmap(r => DistributionTypes.SampleSet(r))
-  }
-
-  /*
+  module InputValidator = {
+    /*
      It would be good to also do a check to make sure that probability mass for the second
      operand, at value 1.0, is 0 (or approximately 0). However, we'd ideally want to check 
      that both the probability mass and the probability density are greater than zero.
      Right now we don't yet have a way of getting probability mass, so I'll leave this for later.
  */
-  let getLogarithmInputError = (t1: t, t2: t, ~toPointSetFn: toPointSetFn): option<error> => {
-    let firstOperandIsGreaterThanZero =
-      toFloatOperation(
-        t1,
-        ~toPointSetFn,
-        ~distToFloatOperation=#Cdf(MagicNumbers.Epsilon.ten),
-      ) |> E.R.fmap(r => r > 0.)
-    let secondOperandIsGreaterThanZero =
-      toFloatOperation(
-        t2,
-        ~toPointSetFn,
-        ~distToFloatOperation=#Cdf(MagicNumbers.Epsilon.ten),
-      ) |> E.R.fmap(r => r > 0.)
-    let items = E.A.R.firstErrorOrOpen([
-      firstOperandIsGreaterThanZero,
-      secondOperandIsGreaterThanZero,
-    ])
-    switch items {
-    | Error(r) => Some(r)
-    | Ok([true, _]) =>
-      Some(LogarithmOfDistributionError("First input must completely greater than 0"))
-    | Ok([false, true]) =>
-      Some(LogarithmOfDistributionError("Second input must completely greater than 0"))
-    | Ok([false, false]) => None
-    | Ok(_) => Some(Unreachable)
+    let getLogarithmInputError = (t1: t, t2: t, ~toPointSetFn: toPointSetFn): option<error> => {
+      let firstOperandIsGreaterThanZero =
+        toFloatOperation(
+          t1,
+          ~toPointSetFn,
+          ~distToFloatOperation=#Cdf(MagicNumbers.Epsilon.ten),
+        ) |> E.R.fmap(r => r > 0.)
+      let secondOperandIsGreaterThanZero =
+        toFloatOperation(
+          t2,
+          ~toPointSetFn,
+          ~distToFloatOperation=#Cdf(MagicNumbers.Epsilon.ten),
+        ) |> E.R.fmap(r => r > 0.)
+      let items = E.A.R.firstErrorOrOpen([
+        firstOperandIsGreaterThanZero,
+        secondOperandIsGreaterThanZero,
+      ])
+      switch items {
+      | Error(r) => Some(r)
+      | Ok([true, _]) =>
+        Some(LogarithmOfDistributionError("First input must completely greater than 0"))
```

Comment:
`must be completely` or `must completely be`

Reply:
good point